### PR TITLE
CHE-6629 Add openshift environment manager implementation

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Add "openshift" environment manager implementation

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/6629
https://github.com/eclipse/che/issues/6868

### CODE REVIEW

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
@olexii4 FYI there is merging conflict

File: dashboard/src/components/api/recipe/che-recipe.factory.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  create(recipe: che.IRecipe): ng.resource.IResource<T>;
+  getRecipes(): ng.resource.IResource<T>;
+}
+
+/**
+ * This class is handling the recipes retrieval
+ * It sets to the array of recipes
+ * @author Florent Benoit
+ */
+export class CheRecipe {
```
Comment (ashumilova) [inline dashboard/src/components/api/recipe/che-recipe.factory.ts]: on API side recipe service was removed - can be removed here as well.

File: dashboard/src/components/api/environment/openshift-environment-manager.ts [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+import {EnvironmentManager} from './environment-manager';
+import {IEnvironmentManagerMachine} from './environment-manager-machine';
+import {OpenshiftParser, IPodList, IPodItem, IPodItemContainer} from './openshift-parser';
+import {CheRecipeTypes} from '../recipe/che-recipe-types';
+
+enum MemoryUnit { 'B', 'Ki', 'Mi', 'Gi' }
+
+
+/**
+ * This is the implementation of environment manager that handles the docker compose format.
```
Comment (ashumilova) [inline dashboard/src/components/api/environment/openshift-environment-manager.ts]: that handles the docker compose format -> that handles the openshift format

File: dashboard/src/app/workspaces/workspace-details/che-recipe.service.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    return this.isCompose(recipe) || this.isOpenshift(recipe);
+  }
+
+  /**
+   * Returns true if the recipe type is a known type.
+   *
+   * @param {che.IRecipe} recipe
+   * @returns {boolean}
+   */
+  isKnownType(recipe: che.IRecipe): boolean {
```
[the remark this answers is not in the document]
Comment (olexii4) [reply dashboard/src/app/workspaces/workspace-details/che-recipe.service.ts]: Ok

File: dashboard/src/components/api/recipe/che-recipe.factory.ts (hunk 1 above)
Comment (olexii4) [reply dashboard/src/components/api/recipe/che-recipe.factory.ts]: done

**Review by akurinnoy — APPROVED** [review]

**Review by ashumilova — APPROVED** [review]

**Pull request merged**